config/stack.go

```diff
@@ -0,0 +1,111 @@
+package config
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/gruntwork-io/terragrunt/config/hclparse"
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+// StackConfigFile represents the structure of terragrunt.stack.hcl stack file.
+type StackConfigFile struct {
+	Locals *terragruntLocal `hcl:"locals,block"`
+	Units  []*Unit          `hcl:"unit,block"`
+}
+
+// Unit represent unit from stack file.
+type Unit struct {
+	Name   string `hcl:",label"`
+	Source string `hcl:"source,attr"`
+	Path   string `hcl:"path,attr"`
+}
+
+// ReadStackConfigFile reads the terragrunt.stack.hcl file.
+func ReadStackConfigFile(ctx context.Context, opts *options.TerragruntOptions) (*StackConfigFile, error) {
+	opts.Logger.Debugf("Reading Terragrunt stack config file at %s", opts.TerragruntStackConfigPath)
+
+	parser := NewParsingContext(ctx, opts)
+
+	file, err := hclparse.NewParser(parser.ParserOptions...).ParseFromFile(opts.TerragruntStackConfigPath)
+	if err != nil {
+		return nil, errors.New(err)
+	}
+	//nolint:contextcheck
+	if err := processLocals(parser, opts, file); err != nil {
+		return nil, errors.New(err)
+	}
+	//nolint:contextcheck
+	evalParsingContext, err := createTerragruntEvalContext(parser, file.ConfigPath)
+	if err != nil {
+		return nil, errors.New(err)
+	}
+
+	config := &StackConfigFile{}
+	if err := file.Decode(config, evalParsingContext); err != nil {
+		return nil, errors.New(err)
+	}
+
+	return config, nil
+}
+
+func processLocals(parser *ParsingContext, opts *options.TerragruntOptions, file *hclparse.File) error {
+	localsBlock, err := file.Blocks(MetadataLocals, false)
+
+	if err != nil {
+		return errors.New(err)
+	}
+
+	if len(localsBlock) == 0 {
+		return nil
+	}
+
+	if len(localsBlock) > 1 {
+		return errors.New(fmt.Sprintf("only one locals block is allowed %s stack file, "+
```

Comment:
Let's change this to:

"Only one locals block is allowed per stack file, but found %d in %s"